reporting_app/__init__.py

```diff
@@ -31,34 +93,34 @@ def pipeline_report(pipeline_type, view_type):
     endpoint = endpoints[pipeline_type]
 
     if view_type == 'all':
-        query = rest_query(endpoint)
+        query = rest_api().api_url(endpoint)
     elif view_type in statuses:
-        query = rest_query(endpoint, match={'$or': [{'proc_status': s} for s in statuses[view_type]]})
+        query = rest_api().api_url(endpoint, match={'$or': [{'proc_status': s} for s in statuses[view_type]]})
     else:
         fl.abort(404)
         return None
 
-    return fl.render_template(
+    title = util.capitalise(view_type) + ' ' + pipeline_type
+
+    return render_template(
         'untabbed_datatables.html',
-        table=datatable_cfg(
-            util.capitalise(view_type) + ' ' + pipeline_type,
-            pipeline_type,
-            query
-        )
+        title,
+        table=datatable_cfg(title, pipeline_type, query)
     )
 
 
-@app.route('/runs/<run_id>')
+@app.route('/run/<run_id>')
+@flask_login.login_required
```

Comment:
There used to be an overall `/runs/` endpoint that listed all runs, so it made sense to use 'runs' in the plural. Now, however, that is done by `/pipelines/runs/` and this endpoint will now always be used to visit a single run called `<run_id>`, so the plural is less relevant now.

This would be a breaking change for quite a small reason, though, so we can reverse it if we don't think it's a good idea.